R/assert.R

```diff
@@ -31,6 +31,11 @@
 vec_assert <- function(x, ptype = NULL, size = NULL, arg = NULL) {
   arg <- arg %||% as_label(substitute(x))
 
+  if (!vec_is_vector(x)) {
+    msg <- paste0("`", arg, "` must be a vector, not a scalar")
```

Comment:
Can we replace scalar with something more specific? Maybe, "not a `class(x)[[1]]`"?

Reply:
This is an error that users will see, and I think "scalar" will be too confusing because they won't be familiar with our precise definition.

Reply:
I included `friendly_type_of()` which we already use in rlang and purrr. I'll export it from rlang at the next release: https://github.com/r-lib/rlang/issues/458